Update mod icon colors

Cwazywierdo opened the pull request:
Closes #19002

Updates `ModIcon.backgroundColour` case values to the same ones found in `OsuColour.ForModType`.
As a note, this changes the color scheme from the one defined in osu-web, to the "basic color scheme" defined in the figma. As such, the differences between background and highlight color become more apparent.

Cwazywierdo committed:
Update mod icon colors

peppy commented:
Pretty sure you want to replace this whole thing with a call to the `ForModType` method you mentioned. The highlighted colour is likely not used by anything anymore, but you will need to confirm that.

Cwazywierdo commented:
The highlighted color is being used in the `LeaderboardModSelector` in `BeatmapSetOverlay`.

Cwazywierdo committed:
Replace switch statement with ForModType

In order to make `highlightedColour` dependent on the mod type color, the color is converted to an `osu.Framework.Graphics.Colour4` and calls `Lighten`.

peppy commented on osu.Game/Rulesets/UI/ModIcon.cs:
```diff
@@ -123,40 +123,12 @@ private void updateMod(IMod value)
+            backgroundColour = typeColour;
+            highlightedColour = ((Colour4)typeColour).Lighten(.2f);
+
+            if (value.Type == ModType.System)
+                modIcon.Colour = colours.Yellow;
```
I do wonder if this can be moved to the `ForModType` return?

@bdach do you recall why it's specified as `Gray` there and if that ever shows up?

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Graphics/OsuColour.cs#L186-L187

peppy replied:
I've removed the yellow local colour logic for now. Using grey feels fine for the time being.

peppy committed:
Add test showing all mod icons for reference

peppy committed:
Simplify colour assigning logic and remove system mod colour for now

peppy committed:
Merge branch 'master' into mod-icon-colors
